new governance flag

Vitomir2 opened the pull request:
# Description

- add a new mandatory flag for the governance address;
- apply the value to be the governance for the core contracts;
- update the polybft_setup docs;

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Used the genesis command to generate the genesis file, run the local blockchain, and test if the correct address is the governance.

Vitomir2 committed:
new governance flag

add a new mandatory flag for the governance address;
apply the value to be the governance for the core contracts;
update the polybft_setup docs;

R-Santev commented on command/genesis/params.go:
```diff
@@ -598,3 +605,20 @@ func (p *genesisParams) getResult() command.CommandResult {
 	}
 }
+
+func (p *genesisParams) validateGovernanceAddress() error {
```
add check isAddress

R-Santev replied:
add check to ensure it is not the same address as the proxyContractsAdmin

R-Santev commented on h_docs/polybft_setup.md:
````diff
@@ -32,10 +32,10 @@ I am describing our custom process, because it is different.
 
 ```
-./hydra genesis --block-gas-limit 10000000 --epoch-size 10 --validators-path ./ --validators-prefix test-chain- --consensus polybft --native-token-config Hydra:HDR:18:true:0x211881Bb4893dd733825A2D97e48bFc38cc70a0c --premine 0x211881Bb4893dd733825A2D97e48bFc38cc70a0c:70000000000000000000000 --premine 0xdC3312E368A178e24850C6dAC169646c5fD14b93:30000000000000000000000 --proxy-contracts-admin 0x2118 ...
+./hydra genesis --block-gas-limit 10000000 --epoch-size 10 --validators-path ./ --validators-prefix test-chain- --consensus polybft --native-token-config Hydra:HDR:18:true:0x211881Bb4893dd733825A2D97e48bFc38cc70a0c --premine 0x211881Bb4893dd733825A2D97e48bFc38cc70a0c:70000000000000000000000 --premine 0xdC3312E368A178e24850C6dAC169646c5fD14b93:30000000000000000000000 --proxy-contracts-admin 0x2118 ...
````
```suggestion
./hydra genesis --block-gas-limit 10000000 --epoch-size 10 --validators-path ./ --validators-prefix test-chain- --consensus polybft --native-token-config Hydra:HDR:18:true:0x211881Bb4893dd733825A2D97e48bFc38cc70a0c --premine 0x211881Bb4893dd733825A2D97e48bFc38cc70a0c:70000000000000000000000 --premine 0xdC3312E368A178e24850C6dAC169646c5fD14b93:30000000000000000000000 --proxy-contracts-admin 0x211881Bb4893dd733825A2D97e48bFc38cc70a0c --governance <governance-address-here> --chain-id 8844
```

Vitomir2 committed:
update

delete the validateProxyContractsAdmin and make a general helper function in the commands that checks if address is valid;
validate that the governance address != proxy contracts admin;
update docs;

Vitomir2 committed:
updates

Vitomir2 committed:
update
